### PR TITLE
string_utils: provide a version of strchrnul() in case it's not avail…

### PR DESCRIPTION
**Pull request by brauner**
…able

This should only happen on Android.

Signed-off-by: Christian Brauner <christian.brauner@ubuntu.com>

### CODE REVIEW

File: src/lxc/string_utils.c [hunk 1]
```diff
@@ -1028,3 +1028,135 @@ void remove_trailing_slashes(char *p)
 	while (--l >= 0 && (p[l] == '/' || p[l] == '\n'))
 		p[l] = '\0';
 }
+
+#ifndef HAVE_STRCHRNUL
+/* Taken from glibc. */
```
Comment (stgraber) [inline src/lxc/string_utils.c:1033]: For this kind of stuff we have `lxc/include/` where we track where things come from and their license.

Comment (brauner) [reply src/lxc/string_utils.c:1033]: Ugh, I was trying to avoid that shenanigans...

Comment (stgraber) [reply src/lxc/string_utils.c:1033]: Yeah but `taken from glibc` while our licenses are compatible is a bit light on details for proper tracking of code origin :)

Comment (brauner) [reply src/lxc/string_utils.c]: Done.

**Pull request merged**